### ISSUE TITLE
Load cell and motor firmware

### ISSUE THREAD

**Issue by Phyntox** [toastfr35/KhordUino#2]
Hi. I have a problem getting the readings from the load cell. I think I connected everithing right.
Also, I have problems running the motor. Nothing seems to happen.

I am using the Arduino UNO. Maybe this is the problem. All the buttons are working.

Could you please share the wiring diagram (more detailed like the DM542 wiring..)

Also it would be nice to have a cancle or back button when you are in the set and pull option.

And a reset button maybe  (Maybe combination of long press of two buttons)?

**Comment by klwill1192** [comment]
Can you tell me which luggage scale you bought to get the load cell?  I've bought a couple and neither was the right one for this project.  Thank you.

**Comment by Phyntox** [comment]
Any load cell should work, but i thik the problem is that the code is tailored to one type. I'm tearing the code back to the raw reading of the cell and I'll try to "calibrate" the reading to get something useful out

**Comment by Phyntox** [comment]
I think i somehow calibrated the load cell inside the code.

Now i just need to see the actual wiring of the DM542. Can someone share a diagram or picture?

**Comment by klwill1192** [comment]
I'm still a little ways away from wiring up the badminton stringer, but
here are some pics from another project I designed that also used a DM542
and stepper motor.  You can see how the stepper motor connects to the DM542
here.  If the motor spins the reverse of what you expect, I believe
swapping the two inner or two outer wires will fix the problem.

Here is the original project:  https://imgur.com/gallery/xcwB9ku

[image: IMG_20231230_201622__01.jpg]
[image: IMG_20231230_201440__01.jpg][image: IMG_20231230_201528__01.jpg]

**Comment by klwill1192** [comment]
> I think i somehow calibrated the load cell inside the code.
> 
> Now i just need to see the actual wiring of the DM542. Can someone share a diagram or picture?

I'm still a little ways away from wiring up the badminton stringer, but
here are some pics from another project I designed that also used a DM542
and stepper motor.  You can see how the stepper motor connects to the DM542
here.  If the motor spins the reverse of what you expect, I believe
swapping the two inner or two outer wires will fix the problem.

Here is the original project:  https://imgur.com/gallery/xcwB9ku

![IMG_20231230_201440__01](https://github.com/toastfr35/KhordUino/assets/29135360/013e933d-1241-4995-94e4-e4e98973f246)
![IMG_20231230_201528__01](https://github.com/toastfr35/KhordUino/assets/29135360/9cdb4421-9bb5-4654-bad4-69ec00430911)
![IMG_20231230_201622__01](https://github.com/toastfr35/KhordUino/assets/29135360/704a4c96-1bb8-45be-bcfb-03869d1e1814)

**Comment by Phyntox** [comment]
I did it. i forgot to wire the GNDto EN-. now it works. I need to fine tune the pulling force.

**Comment by StealthSZN** [comment]
Hello,

I'm new to the 3d print world, and going to attempt this project. Do you guys have any additional recourses/photos/videos that you could share, and would help me out. Reading over the manual, it would be great to have some videos or pictures to go along as I on not to well versed in these types of projects.

I would appreciate it a ton!

**Comment by StealthSZN** [comment]
It would also be great if someone could look over my parts list to make sure everything I am going to order is compatible.

**Comment by Phyntox** [comment]
Buy as it is written in BOM. but expect to do some 3d modelling. My cell was different so i had to make an adapter. And then the calibration of the cell.  And steeper driver... just start and learn and solve problems when they occur. Ill try to help as much as i can 😉

**Comment by StealthSZN** [comment]
That would be awesome, only question I have right now is: do i need an Arduino Breakout board? or do I only need the nano?

Do you have another place I can message you if I have any questions?

Thanks a ton!

**Comment by Phyntox** [comment]
I used arduino Uno. With breakout board. It's easier. And I made some adjustments regarding the main board. I am planning to make ot a bit different so for the concept testing i just 3d printed the bearing and used airsoft balls. It works perfectly (for testing at least)

**Comment by StealthSZN** [comment]
So doable without the breakout board or should I just get one to make it easier?

**Comment by klwill1192** [comment]
> So doable without the breakout board or should I just get one to make it easier?

Both.  I am using a Nano with the breakout board.  It's pretty cheap compared to the total cost of the project, so I think the convenience is worth the expense.

Here's the lazy susan I ordered:  https://www.amazon.com/gp/product/B07T2D8D71

You'll end up drilling out all12 holes, so it doesn't matter if some of them don't go all the way through.  The important thing is that this one had 6 each on both the inner and outer circles.

You probably won't find the bearing that matches the instructions.  I've ordered 10 different luggage scales and none of them were the right type.  I've got four more on the way, but I'm not hopeful.  Fortunately Amazon lets me return the ones I'm not going to use.  I'm currently deciding between modifying one of the luggage scale load cells or purchasing a 30 kg S-type tension load sensor and building my own rig.  Antoine said that V2 will use a different load cell.  It would be nice if he'd give us a sneak peak.  I've reached out through GitHub, but haven't gotten a response yet.

I have also modifiect the PSU box to use an intgrated plug/switch/fuse.  

Antoine also said that he elongated the legs a bit for V2, so I may do that with V1.

One more item on the shopping list:  a cheap racket to test with.  Not going to start out with one of my nicer ones.

**Comment by Phyntox** [comment]
You should not search for the right load cell for the code.  You should adjust the code to any load cell you buy. I made an output of the actual reading of the loadcell. And worked from there. Today i will fine tune it. So find out what are you actually reading.  I was getting numbers around 500000when loaded with 15kg and 400000 without a load.

**Comment by StealthSZN** [comment]
Luckily i have a cheap racket I can use haha.

Thank you for all the info, I really appreciate it!

This is the load cell I was going to order, curious to know if you guys think it'll work: https://shorturl.at/rCEIX

**Comment by klwill1192** [comment]
> This is the load cell I was going to order, curious to know if you guys think it'll work: https://shorturl.at/rCEIX

Whether or not it will work depends on what's inside the case.  The issue isn't so much the load cell as the hardware that connects to the two sides, and whether it can be adapted to connect between  the gripper and tensioner.

I'm seriously thinking about abandoning the cheap luggage scales and redesign the parts to use this:  https://www.amazon.com/dp/B0CPDZ7RS6

**Comment by StealthSZN** [comment]
hmm yeah I knew it might be an issue. I was debating to either attempt to build it now, or wait for V2. I'm only about halfway done printing all the parts. I would be of no help trying to redesign the tensioning unit, but if you figure out a better way and would be willing to share, that would be amazing.

**Comment by Phyntox** [comment]
![20240104_215043_1_1](https://github.com/toastfr35/KhordUino/assets/155113902/0f3e3eec-31ca-44e1-aad6-ab1526337605)

**Comment by StealthSZN** [comment]
Looks awesome, how were you able to get it to work? just changing the code?

Can you convert it into lbs?

**Comment by Phyntox** [comment]
`/********************************************************/
/*                  TENSION                             */
/********************************************************/

/*********************************
 * Display the current tension immediately
 *********************************/ 
void show_tension_now() 
{
  lcd.setCursor(0, 0);
  ///////lcd.print(loadcell.read()); // for testing the output measurement
  lcd.print("     " + String(tension_kg / 10) + "." + String(tension_kg % 10) + " kg    ");
}

/*********************************
 * Display the current tension at most every 0.5 s
 *********************************/ 
void show_tension()
{
  static unsigned long last = 0;
  unsigned long now = millis();
  unsigned long dt = now - last;
  if (dt > 500) {
    show_tension_now();
    last = now;
  }
}

/*********************************
 * Read load cell and update current tension
 *********************************/ 
void update_tension() 
{
  long v;
  if (loadcell.is_ready()) {
    v = ((((loadcell.read()-initialReading))/factor)-11); 
    if (v < 0) v = 0;  //if negative number, shows 0
    tension_int = v;
    tension_kg = v;
    if (tension_display_enabled) show_tension_now();
  }
}
`
This is my code  for tension if it helps. the tare didnt work for some reason so i took 20 average at the beginning and subtract it in the formula, then i calculated the factor (which in my case is 5900) and subtracted 11 (as 1.1kg, to increase final tension)

**Comment by Phyntox** [comment]
yes, it can be converted to lbs.
you should calibrate the cell anyway so the numerical readout doesnt really matter.
And in the code press ctrl+H and replace the kg for lbs... but in my case I converted lbs to kg.

**Comment by StealthSZN** [comment]
awesome thank you!

**Comment by Phyntox** [comment]
I've made quiet a lot of changes. But making it easy for anyone to calibrate the cell is complex. Doable, but unfortunately not with my low knowledge of the code.

**Comment by StealthSZN** [comment]
I have basic understanding of python and some other languages. I've never done anything with arduino.

Is your machine up and running? @Phyntox

**Comment by Phyntox** [comment]
Hint: when calibrating, first find the factor that makes all the reading through the tension spectre has the same error. So for example when adjusting to 35 lbs the actual tension mesured with external scale should be 30 and when stringing with 25 the actual 20... after that, you just add or subtract the difference of 5 lbs for example...

**Comment by StealthSZN** [comment]
So you're saying in your case it was 5lbs but mine may be different depending on the cell?

**Comment by Phyntox** [comment]
Yes I strung one racket at 10 kg, just to proof myself if it is worth developing. Now that i have done that succesfully, I will upgrade the frame gripping mechanism made from some Alu 20x20 profiles. and later I will probbably develope some new string gripping mechanism.
The thing that botheres me the most is the tension check rate, which is causing the stopping of the motor while tensioning.
Increasing it's value, does nothing. the check rate just stays at some frequency, that is either way anoying. and my understanding of the programming is not sufficient for this troubleshooting.
I tried pasting the video with sound but was too big.

**Comment by Phyntox** [comment]
> So you're saying in your case it was 5lbs but mine may be different depending on the cell?

Yes in my case the actual reading was 1,1 kg to low, so I adjusted it.

**Comment by StealthSZN** [comment]
Hmmm, i might just wait until he releases V2 or until someone creates a design with less room for error. 

Thank you both for the info, I may print the rest of the parts and attempt it soon and if I've worked up the courage!

If you have any other way I can contact you let me know, I can give out my instagram or something like that. I'd love to see pictures/videos with it actually stringing a racket!

**Comment by Phyntox** [comment]
It's really easy.  Instead of the code that shows the tension on led, you paste only load cell read.  And you go to test and see what values are you getting.. then work from there.

**Comment by klwill1192** [comment]
I've decided to go in a slightly different direction from the cheap luggage scales.  I just ordered the 30kg version of this load cell:  https://www.aliexpress.us/item/3256805789835547.html

And a selection of these M6 threaded T handles:  https://www.aliexpress.us/item/3256805500831086.html

I think I'll end up using the M6x25mm and M6x50mm, but I bought several sizes to play it safe.  I've already modeled the GS_springs_housing and TS_body in Fusion, and I'll make the necessary changes to accommodate the load sensor and appropriate T handles once they arrive.

**Comment by StealthSZN** [comment]
That's awesome, keep me updated with results!

**Comment by klwill1192** [comment]
Along with a dozen scales from Amazon (none of which had the right type of load sensor for this project), I also ordered a few from Temu.  Lo and behold, two of three had the right innards.  Still waiting for my Aliexpress load sensor and threaded T-handles to arrive, but in the meantime I have the right load sensor for the original design to play with--plus a spare, in case anyone is interested.

**Comment by Phyntox** [comment]
Again.  Don't go buying the load cells too fit the code.  Adjust the code and any load cell will work

**Comment by Phyntox** [comment]
It's just a matter of calibration

**Comment by StealthSZN** [comment]
Nice! Hopefully the rest of the way is smooth sailing. Was it the one you linked earlier?

**Comment by klwill1192** [comment]
> Again. Don't go buying the load cells too fit the code. Adjust the code and any load cell will work

I was shopping for load cells to fit the existing 3D-printed hardware.  But before these arrived I had already decided to order a different load cell and hardware.  I'm still waiting for delivery of that other load cell, but I now have something to test with before the new load cell arrives.  Once it does, I'll redesign the 3D-printed parts to fit.

**Comment by Phyntox** [comment]
Please let me know if your motor is also stopping while reading the tension.

**Comment by magoralczyk** [comment]
Hi, it is a few months later now. How is the development going on version 2? Is there a release on the horizon?
